### DISCUSSION TITLE
Please don't use LibAdwaita library

### DISCUSSION THREAD

**Discussion by medmedin2014** [xournalpp/xournalpp#5301] [General]
### Is Your Feature Request Related to a Problem?

LibAdwaita is meant for developing apps to run only on GNOME desktop, they forced that Adwaita theme, and it's difficult to change it on other non GNOME desktop. This app is used on multiple desktops like Plasma, XFCE... and any LibAdwaita app has serious problem to integrate with other native apps.
https://discuss.kde.org/t/any-plasma-hack-fix-to-libadwaita-look/2946

### Feature Description

Please use pure GTK widgets and don't lock this great app into GNOME trap.

**Comment by rolandlo** [comment]
That's definitely a discussion we should have now and discuss pros and cons. 

Cons:
- LibAdwaita apps may look alien on Plasma and other non-Gnome desktops as they can hardly be themed in a way to fit the particular desktop
- LibAdwaita is not available on old Linux distros natively (which would force users of those distros to use the flatpak or snap package)

Pros:
- LibAdwaita helps tremendously in making apps adaptive, which simplifies running those apps on devices with different form factors (including Linux tablets, e-readers or smartphones)
- LibAdwaita widgets support various gestures out of the box, which would be time-consuming and error-prone to implement by hand
- It's much easier to style an app consistently using LibAdwaita than without it. In Xournal++ there are tons of inconsistencies in the UI style
- For users that are acquainted with LibAdwaita apps it's easier to learn to use another LibAdwaita app than a random Gtk app
- Although pure Gtk 4 apps are easier to theme in accordance with a non-Gnome desktop, the theming is untested by the developers and may break lots of stuff
- In future on Gnome desktops non-LibAdwaita apps may start looking alien as well...
- There is no way to create an app with Gtk that looks on all supported platforms (including Windows and MacOS) completely at home. We would have to separate the core logics of our app into a library and write different apps for different desktops to have native looking apps on all desktops (and unfortunately that's a ton of work).

So far I have read various comments which would love for Xournal++ to be redesigned as a LibAdwaita app and you're the first one arguing against it. What does the majority of users think?! Please comment below!

See #4561, #4339 and #2722 for related issues/discussions.

Personally I would strongly advocate for using LibAdwaita. It makes things a lot easier from the developer point of view. I love the looks of the modern Gnome desktop and think that even though LibAdwaita apps may look alien on non-Gnome desktops, it's better than to have a less consistent and less powerful app.

**Reply by Secret-chest** [reply]
Non-GNOME desktops are also MATE, Cinnamon, Xfce.

**Comment by medmedin2014** [comment]
I have nothing against LibAdwaita as a technology, but the people behind it tend to push the library to fit and fix GNOME only problems without caring about other desktops. They refuse to provide a standard API to change the embedded Adwaita theme, which means any attempt from any major Linux desktop to customize a theme for LibAdwaita will be broken with any major upgrade. Currently, they only officially support only some sort of accent color.

Even GNOME devs describe LibAdwaita as a library to create apps for GNOME desktop and not for Linux.

**Comment by medmedin2014** [comment]
The same problem happened to Manjaro distribution which officially supports multiple desktops (Plasma, GNOME, XFCE) and other community flavors like (Cinnamon, MATE and Budgie), they upgraded their software manager Pamac to GTK4, but the feedback was bad due to bad integration of LibAdwaita with multiple desktops, at the end they separated the app to two branches one continuing to use GTK3 and another one for GNOME using GTK4.
https://forum.manjaro.org/t/pamac-not-following-system-themes-plasma/144244

**Comment by bhennion** [comment]
I didn't realize LibAdwaita had this limitation, and it is definitely so on purpose: [stopthemingmy.app](https://stopthemingmy.app/).

Now, I feel it's important to keep the app independent and not make it a GNOME app. Besides the question of LibAdwaita, I think an app like xournal++ is fairly different than most apps. It's primary task is writing with a pen. I would say what the app should focus on is:

1. Writing.
2. Easy access to a large number of tools (so toolbars, toolboxes, menus or things of the sort).

Those tasks are closer to the requirements of drawing or multimedia-editing apps, and those requirements (especially 2.) are pretty much antipodal to Gnome's and Adwaita's lead towards sparse UIs. I haven't found any LibAdwaita app example with similar needs - Gimp is GTK2, Inkscape is GTK3, Gnome Circle's Drawing is GTK3.

Another remark: being overly adaptative is useless for us. I doubt we can manage a production-ready UI for phone-sized devices. Writing with a pen on a phone is not handy at all, independently of the software.

I've spend some time playing around LibAdwaita, and as far as I can tell:

1. Our core functionalities are not going to be improved nor impaired by LibAdwaita. LibAdwaita does not provide better pen support, and LibAdwaita does not provide anything toolbox-oriented. We can of course rethink the static toolbar idea but I haven't seen anything helping for this in Adwaita either.
2. The difference will mostly be in the dialogs. The dialogs are precisely something quite separated from the main app, and have different frontends for them would be relatively easy. They all have a distinct class already.

In my opinion, the main window should be maintained as a GtkApplicationWindow, and its content kept pure gtk, simply because the benefits of LibAdwaita there don't seem to outweight the drawbacks.


In the long run, separating the core business from the "frontend" in the codebase is a great goal. We already made a lot of progress in this direction (most widget interactions now go through one single class: the ActionDatabase) but a lot remains.
The port to GTK4 is an opportunity to make more progress on this question: the main widget and its input system needs to be entirely rewritten. Writing it as a pair backend/frontend seems like the right thing to do.
And since we need to port to GTK4 before even being able to use LibAdwaita...

**Comment by rolandlo** [comment]
> Those tasks are closer to the requirements of drawing or multimedia-editing apps, and those requirements (especially 2.) are pretty much antipodal to Gnome's and Adwaita's lead towards sparse UIs. I haven't found any LibAdwaita app example with similar needs - Gimp is GTK2, Inkscape is GTK3, Gnome Circle's Drawing is GTK3.

Take a look at [Rnote](https://flathub.org/apps/com.github.flxzt.rnote). It is written in Rust, but that doesn't make much difference. It uses all the feature of LibAdwaita and the UI is pretty well thought out. It has exactly the same needs as Xournal++ does.


> 1. Our core functionalities are not going to be improved nor impaired by LibAdwaita. LibAdwaita does not provide better pen support, and LibAdwaita does not provide anything toolbox-oriented. We can of course rethink the static toolbar idea but I haven't seen anything helping for this in Adwaita either.

While I agree that pen support is not affected by LibAdwaita, it does offer some touch support that pure Gtk 4 widgets don't have. Since users of a stylus-based note-taking app usually also have a touchscreen, it looks reasonable to support touch well. For instance the AdwOverlaySplitView supports an edge swipe gesture for showing the sidebar, and a swipe from the sidebar for hiding it. 
When working with multiple documents (in future) AdwTabBar and AdwTabView would be nice to have.
Also there is e.g. AdwSplitButton, which might be handy for a custom pen/highlighter tool like I designed in https://github.com/xournalpp/xournalpp/discussions/4561#discussioncomment-7138492


> 2\. The difference will mostly be in the dialogs. The dialogs are precisely something quite separated from the main app, and have different frontends for them would be relatively easy. They all have a distinct class already.

I agree that the dialogs benefit the most from Libadwaita. See e.g. the preferences dialog https://github.com/rolandlo/xournalpp-libadwaita-mockup/blob/main/preferences/main.blp, which becomes much easier to write than with pure Gtk. It's almost completely separated from the code.


> In the long run, separating the core business from the "frontend" in the codebase is a great goal. We already made a lot of progress in this direction (most widget interactions now go through one single class: the ActionDatabase) but a lot remains.
> The port to GTK4 is an opportunity to make more progress on this question: the main widget and its input system needs to be entirely rewritten. Writing it as a pair backend/frontend seems like the right thing to do.

I fully agree with that.

**Comment by rolandlo** [comment]
> Another remark: being overly adaptative is useless for us. I doubt we can manage a production-ready UI for phone-sized devices. Writing with a pen on a phone is not handy at all, independently of the software.

While writing on a phone is not handy, a phone (or similar device) might still be useful for a presentation on a projector/external monitor or maybe for signing a document.

**Comment by denyskon** [comment]
A somebody who used Xournal++ on Windows and on Linux for many years now, I strongly support using LibAdwaita.

The pros I see in it are:
- consistent & pretty UI
- properly built widgets that can simply be used without reinventing the wheel
- responsiveness, gestures and similar - that's also the reason newer GNOME apps are by far the most usable on touchscreens
- same look & feel on all systems & desktop environments: as hard as you try, GTK apps will never look good when you force them to use your Plasma theme or similar, and you'll never achieve native UI on Windows or Mac. With LibAdwaita you at least get pretty UI everywhere which doesn't break because of some icon theme not working well with Xournalpp's custom widgets and similar issues

I'm sure there are more reasons to use it, and tbh I don't really see any major drawbacks

**Reply by medmedin2014** [reply]
The problem is that if any issue is found with the app on non GNOME desktops or on another OS, GNOME devs who control LibAdwaita development will never fix it because they clearly claim that LibAdwaita is a library to only create GNOME apps.

That's the prime reason why KDE devs never attempted to create a temporary theme for LibAwaita, because they are sure that GNOME devs will lock or change their API even more to make apps created on top of LibAdwaita usable only on GNOME desktop.

And it's also the sole reason why LibAdwaita library will never be merged with GTK4 dev branch.

**Reply by denyskon** [reply]
There is a difference between fixing issues and trying to implement a hacky workaround to change UI even if it is intended not to be changed. And the second one is the reason we have a long history of broken cross-desktop UI in the first place.

**Reply by medmedin2014** [reply]
Using pure GTK4 widgets would be much better than using that locked library. Because having a homogenous app that blends well with other system apps is a must for many end users. I'm long Plasma user and even if some LibAdwaita apps are good and do the job, I hate how they look alien and out of place inside my desktop and always end by removing them and look for suitable alternative.

**Reply by rene-coty** [reply]
As a lambda Linux user I don't see using libadwaita as a drawback in terms of consistency: Qt apps may also look out of place and broken on non-KDE desktops (e.g. on Fedora although it ships different desktop environments and has a solid common base accross its spins) even if they don't embed something similar to libadwaita.. So in terms of UX libadwaita isn't worse than any other UI toolkit in terms of overall user experience, and I'd rather have an app that doesn't feel 'consistent' with the rest of my desktop environment than an app that is unusable under certain circumstances with missing icons or that misses the points in terms of usability on touch screens, adaptivity or ease of use (and that's probably what the end users will most appreciate).

So if for developers using libadwaita represents an advantage in terms of design or time saving, I don't see why we shouldn't take the plunge..

**Reply by medmedin2014** [reply]
> So if for developers using libadwaita represents an advantage in terms of design or time saving, I don't see why we shouldn't take the plunge..

That followed "design" is specific to locked GNOME ecosystem and not suitable to be cross-platform, and to be honest that Adwaita looks quite ugly, and concerning functionalities I don't think there is something that LibAdwaita can achieve which pure GTK4 can't.
Other known GTK apps are already migrating to GTK4 like [Inkscape](https://inkscape.org/news/2023/04/17/inkscape-hiring-accelerating-gtk4-migration/) and I don't think they would approach that cursed library.

**Reply by Secret-chest** [reply]
I don't want "pretty" UI. I want a UI that I can use comfortably; for me that's a lower-contrast blue-grey theme, not white and not black. Would you prefer it looking native on one DE or on tens of DEs?

**Comment by 1over137** [comment]
Customizability is one of the greatest strengths of the Linux desktop. Libadwaita is actively against customization, and refuses to be themed. By using libadwaita for a general purpose application, we are undermining this strength. Compared to libadwaita, Qt doesn't straight out refuse to be themed and can look mostly fitting if you can find an appropriate theme for it, so I think the comparison is bad. I sincerely hope xournalpp would reconsider the decision to use libadwaita. If it has to be used, I hope it should be limited to widgets like toolbars and not include things like client-side decoration that GNOME likes to use, which looks terrible on other desktop environments.

**Reply by rolandlo** [reply]
> Customizability is one of the greatest strengths of the Linux desktop. Libadwaita is actively against customization, and refuses to be themed. By using libadwaita for a general purpose application, we are undermining this strength. Compared to libadwaita, Qt doesn't straight out refuse to be themed and can look mostly fitting if you can find an appropriate theme for it, so I think the comparison is bad. I sincerely hope xournalpp would reconsider the decision to use libadwaita. 

We haven't made a decision yet. Moving from Gtk 3 to Gtk 4 is already a big step and there is a lot we have to learn about it before completing the transition. Libadwaita can only be used with Gtk 4.

> If it has to be used, I hope it should be limited to widgets like toolbars and not include things like client-side decoration that GNOME likes to use, which looks terrible on other desktop environments.

Client side decoration is not exclusively used in Libadwaita apps. In pure Gtk there is the [GtkHeaderBar widget](https://docs.gtk.org/gtk4/class.HeaderBar.html). So would you be opposed to using a GtkHeaderBar as much as using some Libadwaita widgets?

**Reply by 1over137** [reply]
I would not oppose using GtkHeaderBar nearly as much, as it can be themed, like in this image (GTK and Qt are using Arc theme though window decorations are different):
![image](https://github.com/xournalpp/xournalpp/assets/22680475/0d2f2156-a034-42e5-b4c9-87b00a4a762e)
Left: Dolphin (Qt)
Middle: Evolution (GTK, with headerbar, theme is applied)
Right: Dino (GTK + Libadwaita, no theming at all)

I want to say that I actually really like headerbar as a concept. I just don't like how GNOME devs feel a need to enforce the creeping uniformity of their chosen theme and leave no room for customization through libadwaita. I understand the struggle of developing against UI toolkits and the appeal of using a library to simplify it ([My GUI app](https://github.com/FreeLanguageTools/vocabsieve)), but I think using libadwaita is a step too far. We are giving up too much for the benefits it is providing.

**Reply by Secret-chest** [reply]
> In addition, if users want a libadwaita app, there is already RNote. Please leave something for the rest of us non-GNOME users!

LibAdwaita _and_ Rust: perfect for the Fedora trend-chasers that 66% of GNOME users are.

**Comment by medmedin2014** [comment]
> We haven't made a decision yet. Moving from Gtk 3 to Gtk 4 is already a big step and there is a lot we have to learn about it before completing the transition. Libadwaita can only be used with Gtk 4.

I don't know any big/famous and **cross-platform** drawing app that attempted or needed to use LibAdwaita.

- The widely used tablet drawing app Krita uses Qt

- Next Inkscape will use pure GTK4 (without LibAdwaita)

- Next Gimp 3 will use pure GTK3 (without libhandy)

**Reply by denyskon** [reply]
The big difference is that the applications you named don't even try to provide a responsive/touch-friendly interface

**Reply by medmedin2014** [reply]
@denyskon Krita is the most advanced free tablet drawing app in the market, and it doesn't need what you call "responsive/touch-friendly interface" to make it easy for users to complete their professional work.

**Reply by carbon-starlight** [reply]
> @denyskon Krita is the most advanced free tablet drawing app in the market, and it doesn't need what you call "responsive/touch-friendly interface" to make it easy for users to complete their professional work.

@medmedin2014 Except Krita is being used on tablets by artists who also use styluses that may handle clicks as precise as a mouse can handle. Still, it is often being referred as [being hard to get used to](https://www.reddit.com/r/krita/comments/1el52qu/).

**Reply by medmedin2014** [reply]
@carbon-starlight Even the famous audio post-processing app EasyEffects which was built on top of LibAdwaita [is adopting Qt](https://github.com/wwmm/easyeffects/issues/2960), the project was too big to be kept being caged inside GNOME ecosystem, LibAdwaita library is hated on every non GNOME desktop. GNOME devs' silly decisions especially the one forcing that ugly Adwaita theme was the major nail in GNOME coffin, the desktop already [lost the battle](https://pkgstats.archlinux.de/fun/Desktop%20Environments/history) on Arch based systems.

Every GTK app that adopted that cursed LibAdwaita library is getting tons of bugs on non GNOME desktops, that library was designed to lock and force that ugly design on users which contradicts the level of customization that users need on other Linux desktops. It was simply a trap for GTK apps.

**Reply by Secret-chest** [reply]
> > @denyskon Krita is the most advanced free tablet drawing app in the market, and it doesn't need what you call "responsive/touch-friendly interface" to make it easy for users to complete their professional work.
> 
> @medmedin2014 Except Krita is being used on tablets by artists who also use styluses that may handle clicks as precise as a mouse can handle. Still, it is often being referred as [being hard to get used to](https://www.reddit.com/r/krita/comments/1el52qu/).

Isn't it the same with our app? Plus, if you want, you can have a larger theme, which Adwaita makes impossible.

**Reply by carbon-starlight** [reply]
> > > @denyskon Krita is the most advanced free tablet drawing app in the market, and it doesn't need what you call "responsive/touch-friendly interface" to make it easy for users to complete their professional work.
> > 
> > 
> > @medmedin2014 Except Krita is being used on tablets by artists who also use styluses that may handle clicks as precise as a mouse can handle. Still, it is often being referred as [being hard to get used to](https://www.reddit.com/r/krita/comments/1el52qu/).
> 
> Isn't it the same with our app? Plus, if you want, you can have a larger theme, which Adwaita makes impossible.

Well, if you are working on a note-taking app it's always nice to try to make in mobile-friendly. Many people do note-taking on smartphones or tablets without styluses.

**Comment by 1over137** [comment]
On another point, all the libadwaita apps I've used seem to have a serious lack of complex features that greatly reduce their usability. I don't think we can preserve all the features that are currently in xournalpp if a libadwaita-based redesign is to be pursued. Looking at the mockups (https://github.com/xournalpp/xournalpp/discussions/4561), it looks like a lot of features are losing discoverability as GNOME apps don't like menu bars, which I think is a huge drawback. Realisitically, I think people who likes a more minimal app should just use Rnote, and the development effort that would be required to convert xournalpp to a libadwaita app is probably better spent on improving and extending Rnote. It is better for all users if we have a choice between a minimal/friendly/intuitive application and a more complex and powerful one. 

Also, given how libadwaita looks on Linux, I think it would be an even bigger eyesore on Windows where you can't really hack the colorscheme as easily, which could be an important consideration if xournalpp wants to continue supporting Windows.

**Reply by rolandlo** [reply]
@1over137 
Libadwaita apps are usually written for the Gnome desktop and Gnome tends to have simpler apps than e.g. KDE. Please take a look at the mockup in https://github.com/xournalpp/xournalpp/discussions/4561#discussioncomment-8057257. Although not all features have been included yet, it's close to that and will get there (although probably with some removal of duplicated functionality). Do you think discoverability is reduced a lot with that mockup? To me it looks pretty easy to find stuff.

I don't think that using Libadwaita would necessarily increase the development effort. When rewriting GUI elements from scratch it's easier to use Libadwaita than coming up with own GUI elements or a consistent styling. Only it the changes needed for transitioning from Gtk 3 to Gtk 4 are small it's faster to just make those changes.

The largest GUI element we have in Xournal++ (by number of lines) is the preferences dialog. The `settings.glade` file (which doesn't even contain all of the preferences dialog) has currently 6222 lines of xml and there is quite some code accompanying it.
In the redesign attempt (linked above) there are only 1663 lines of ui definitions (split up in 13 files, none larger than 336 lines) and 163 lines of code. It's not 100% complete, but almost. Ok, the comparison is a bit unfair, since the redesign uses blueprint files, which are roughly 20% smaller by line count on average, but still it's roughly a third when comparing them in xml format.

Talking about Color scheme: With Libadwaita's [StyleManager](https://gnome.pages.gitlab.gnome.org/libadwaita/doc/main/class.StyleManager.html) changing from dark theme to light theme (or follow system theme) becomes almost trivial on all platforms. That's something which is not as easy without Libadwaita.

Anyway I would be curious to get some feedback for the mockup linked above. You'll probably find things you like and others you don't like so let us know. You can also try it out yourself, as there are build instructions in the README.

Yes, I believe it's important that Xournal++ behaves nicely with respect to touch. The ideal way to use Xournal++ is with a convertible (which has a touchscreen). The second best option is to use it with a Wacom-(like) tablet, but then you always need to carry two devices with you, which is not great for teaching in different rooms. 
Libadwaita has some widgets with predefined gestures, like for the sidebar in a [OverlaySplitView](https://gnome.pages.gitlab.gnome.org/libadwaita/doc/main/class.OverlaySplitView.html) which you can show/hide via a edge swipe gesture. It's possible to redefine these gestures in Gtk 4, so it's not a deal breaker, but it's additional work to implement the gestures on our own.

**Reply by medmedin2014** [reply]
@rolandlo Truthfully, the old look is much cleaner, beautiful and easy to use on small touch screen (under 15"). That new design is awful and will eat most screen's space with those big buttons on that right sidebar, and having more writing/drawing space is the most feature needed when using any note app.

**Reply by rene-coty** [reply]
@rolandlo Thanks for your feedback! Xournal looks a lot more modern and intuitive with this brand new look! The new tool sidebar seems more clear and easy to use to me, while still looking customizable… good point for discoverabilty in my opinion :) Regarding the space for the canva, there doesn't seem to be any problem to me, as both sidebars can be easily toggled.
Another good point are the buttons in the headerbar: not too many, and they allow you to easily access all the basic options an average user will need without searching the whole UI

**Reply by 1over137** [reply]
@rolandlo 
It doesn't look too bad except for the lack of theming, but the main issue to me is the use of screen space compared to old xournalpp. A major usecase of xournalpp I believe is marking up lecture slides (horizontal) as a form of note taking, and that huge sidebar is going to take up a lot of space on a typical 13 inch convertible laptop. Making it hide and show requires extra interactions, which I think isn't very desirable compared to the current solution.  As far as I can tell, iPad apps don't require extra interactions to change pen color. I think it is safe to assume that user has a pen, and the pen is precise enough to click on small icons like there are now. The main benefit of touch, as I see it, is the ability to zoom and scroll between pages like on several iPad apps, which is a much better experience than clicking on the scrollbar or pressing keys.

I'm still not convinced that the discoverability is nearly as good. Like, I have no idea where I should find "Paper background" or "Open Recent" or "Export as PDF". The fat title bar is also going to take up nearly as much space as the current menubar + toolbar due to GNOME's adwaita theme design, without most of the benefits of the menubar and toolbar. It's a matter of personal taste but there is also too much margins for my liking, compared to most normal desktop programs, which I find objectionable and a waste of limited screen space especially on laptops. The lack of theming in libadwaita makes this worse as you can't as easily adjust the size. In comparison, the current design looks much more utilitarian and uses screen space  efficiently. I think it is a major failing of current UI design paradigms that they want to force your attention on *their* app and not be able to multitask. Xournalpp shouldn't do this.

On Color schemes:
I think a lot of users would prefer their theme to be controlled systemwide, not by individual applications, especially without a way to easily modify that to match system theme. Having a fallback option is fine, though. GTK's existing themes is not just colors and dark or light, but also spacing, transparency, and things like margins.

**Reply by JesusChapman** [reply]
> On another point, all the libadwaita apps I've used seem to have a serious lack of complex features that greatly reduce their usability. I don't think we can preserve all the features that are currently in xournalpp if a libadwaita-based redesign is to be pursued. Looking at the mockups (#4561), it looks like a lot of features are losing discoverability as GNOME apps don't like menu bars, which I think is a huge drawback. Realisitically, I think people who likes a more minimal app should just use Rnote, and the development effort that would be required to convert xournalpp to a libadwaita app is probably better spent on improving and extending Rnote. It is better for all users if we have a choice between a minimal/friendly/intuitive application and a more complex and powerful one.
> 
> Also, given how libadwaita looks on Linux, I think it would be an even bigger eyesore on Windows where you can't really hack the colorscheme as easily, which could be an important consideration if xournalpp wants to continue supporting Windows.

https://gist.github.com/pojntfx/36a1f41ffd34af7d2f3abe43646142e1

https://discourse.gnome.org/t/using-libadwaita-on-linux-windows-apps/29181/3

If possible packaging Libadwaita applications for Windows (and I understand that also for Mac, and if, it also supports the accent color on the Windows theme :))

**Comment by rolandlo** [comment]
Thanks everybody for the feedback. Looks like screen space is a major concern, as is theming ability. I think those are valid arguments. Comparing side by side:
![side_by_side](https://github.com/xournalpp/xournalpp/assets/40485433/757d8673-6224-4999-a903-3018c0380ae8)
it is clear that the new design takes considerably more horizontal space. It does take considerably less vertical space, though. I don't agree with
> The fat title bar is also going to take up nearly as much space as the current menubar + toolbar

It would be possible e.g. to add an option to toggle the headers of the tool sections (like "Drawing area" with the customize and close button) and to make the sidebar be resizeable. The buttons are put into a flowbox, so they would just adapt to the space given by the sidebar. It may also be possible to have some tool groups use the space on the bottom.

About discoverability: "Open Recent" and "Export as PDF" (named "Quick PDF Export" in the new design) were in the menu "File" in the old design and are in the primary menu (the hamburger menu) in the new design. That should be pretty standard.
The "Paper background" is still missing, but it will fit nicely with "Page Template", "Paper Color" and "Paper Format" in the "Document Properties" secondary menu. 

Nobody said anything about the dialogs (and about the custom pen/marker tool). Does that mean that nobody finds issues with how the dialogs look? The UI definition of the main window is much smaller than the UI definition of the preferences dialog (and therefore easier to change).

**Reply by 1over137** [reply]
For the settings dialog I feel like having tabs at the bottom looks rather weird, but I have no other issue with how it looks besides the Adwaita theme, but maybe that's justified for changing settings with only the touchscreen. I'm not that sure how the custom tool is supposed to work, but I think it would be great if it allows you to save a set of pens (blue pen, red pen, yellow highlighter) to be instantly switched and that switching UI should be always present (not hidden on the sidebar).

**Reply by bhennion** [reply]
Thanks a lot for the hard work @rolandlo!
I haven't looked so closely at the dialogs, but I also don't have a strong opinion about them. I think their design is (certainly tedious but) of lesser importance than the main window.
The custom tools are a great addition (also asked for in some ticket somewhere). Discoverablility might be an issue here: how do we create them? It may also be interesting to allow for a custom tool to be "frozen" (so you can't change it anymore, until it's unfrozen).

Overall, I like the sidebar for sifting through tools and discover them, but not really as a daily interface. As others have mentionned, it takes a lot of space.
In order to solve this space question, a simple design (although not as customizable as the current system) may be the following: add a mode where the user pins/unpins tools that are important to them, and add the option to shrink (by swiping for instance) the sidebar into a thin toolbar, containing only the pinned tools.

Your side-by-side screenshot also raises the question: which of vertical space or horizontal space is more important? This depends of course on the user's setup and workflow: screen size obviously, but also portrait or landscape. I guess that in portrait mode, the sidebar would indeed take up way to much h-space.
There are other ergonomy questions I don't have an answer to: what's the most suitable setup for right- vs left-handed, for languages written from right to left or even vertically? The customability of the current version kind of saved us the trouble of those questions so far.

**Reply by Secret-chest** [reply]
I usually use this in portrait orientation and it would make it impossible to use for me.

**Comment by medmedin2014** [comment]
@rolandlo I don't really see what that new design solves compared to this practical layout 

![Screenshot_20240110_175553](https://github.com/xournalpp/xournalpp/assets/7027319/83e684e6-f7b7-4f55-bab0-d446d8aa251a)

What I would have preferred is to see a powerful new layout with tools at right /left/top/bottom side, and when you select a tool you get all its options in toolbar, the same way how Inkscape is doing it


![Screenshot_20240110_175927](https://github.com/xournalpp/xournalpp/assets/7027319/470c54a0-6cc2-48bd-a8c7-f4e92c06e7fb)

![Screenshot_20240110_180112](https://github.com/xournalpp/xournalpp/assets/7027319/ba5de6bf-5719-491a-a5e0-d20aad00c5d2)

Xournalpp is no more a simple tool to take notes, it has potential to add more complex features, so I see no benefits of adopting a GUI library that enforces minimal design and simplicity.

**Comment by JesusChapman** [comment]
<img width="1920" height="1080" alt="imagen" src="https://github.com/user-attachments/assets/0893bce2-4ec2-4b4e-88bd-3e8ffcd20d92" />

I'm sorry, but on my part I totally support the use of Libadwaita for Xournal, Libadwaita is a simple bookstore for developers, modern and very beautiful, customization breaks things and xournal is an application for everyday users, not for geeks that use window managers etc...

Libadwaita is modern, simple, consistent and easy to understand, using it is an excellent option (apart we must take into account the topic of convergence), Gnome adapts well to both desk and phones, an example of my mobile phone executing Gnome:

<img width="1080" height="2220" alt="Captura desde 2025-09-05 19-02-32" src="https://github.com/user-attachments/assets/125f8814-20df-4c75-888f-cdc3f7395da3" />

And finally, if we follow the logic of criticism, KDE applications are also horrible and inons out of KDE, so the case is totally the same. 

But I think that this app must prioritize only two things: be simple and functional. 

And we should not get carried away by people who just want to put a GTK issue, and who continue to use obsolete and unlawful interfaces.

And I extend my hand to Xournal developers if they want someone to test their application, I have a laptop with tactile screen running gnome, and a phone (Google Pixel 3a) running gnome too, in case they want to implement convergence in the app using libadwaita and wish to see how you perform on tactile/mobile devices

**Reply by Secret-chest** [reply]
Hi GSheep. So you want Xournal to only work correctly in GNOME rather than in all desktop environments. The UIs that integrate well aren't obsolete and unlawful, they are simply traditional. I don't like apps to have too much UI design. Apps should not expect any colours, fonts, or other styles.

And also, you are so stupid to assume that users of GTK apps that do not use Adwaita are all on standalone WMs. There is MATE. There is Cinnamon. There is Xfce. In all of these (and GNOME, with the correct libAdwaita backport), a non-libAdwaita UI can integrate very well.

I don't want a Windows-like situation where no two apps look the same.

**Reply by Secret-chest** [reply]
Also, how did you get GNOME on your Pixel 3a?

**Reply by JesusChapman** [reply]
https://postmarketos.org/

**Reply by IanAtSalt** [reply]
Libaidwata is shit. The gnome DE is terrible for usability, and now there's no hope of changing it. These Gnome devs have gone off the deep end with their arrogance!

**Reply by JesusChapman** [reply]
> Libaidwata is shit. The gnome DE is terrible for usability, and now there's no hope of changing it. These Gnome devs have gone off the deep end with their arrogance!

Gnome is very good, simply its design is a different concept. If you like to move everything from your desktop and modify everything you see instead of focusing on your work, Gnome is not for you.

**Reply by medmedin2014** [reply]
> > Libaidwata is shit. The gnome DE is terrible for usability, and now there's no hope of changing it. These Gnome devs have gone off the deep end with their arrogance!
> 
> Gnome is very good, simply its design is a different concept. If you like to move everything from your desktop and modify everything you see instead of focusing on your work, Gnome is not for you.

The history will never forget the crime that those GNOME devs did to the king of Linux desktops, they directed the project to their narrow perspective, and the punishment was truly suitable for them.

**Comment by Secret-chest** [comment]
My use case for Xournal is maximised, on a 12" retina display (Microsoft Surface Pro 7+) with logical resolution of 1368*912. The new design will make it very hard to use the app for me, because I'd have about a quarter of its width taken up by the sidebar. Also, I can forget about using Xournal in portrait.

**Reply by JesusChapman** [reply]
> My use case for Xournal is maximised, on a 12" retina display (Microsoft Surface Pro 7+) with logical resolution of 1368*912. The new design will make it very hard to use the app for me, because I'd have about a quarter of its width taken up by the sidebar. Also, I can forget about using Xournal in portrait.

You are flatly mistaken, Libadwaita is even designed to offer a better experience in cases like yours, your widgets can be adapted to circumstances, for example, this application to edit screenshots changes its way of working automatically by being narrower.

<img width="2606" height="1642" alt="Captura desde 2025-09-09 10-02-44" src="https://github.com/user-attachments/assets/3da6c475-93b6-48a0-9ca7-553aa369164c" />
<img width="964" height="1486" alt="Captura desde 2025-09-09 10-34-37" src="https://github.com/user-attachments/assets/87c23a63-a194-4295-be14-d2d3b45d5e08" />
<img width="992" height="1478" alt="Captura desde 2025-09-09 10-04-19" src="https://github.com/user-attachments/assets/75d9d429-c5a3-4062-9817-0997c3a2d9cc" />

**Reply by Secret-chest** [reply]
I don't need a hamburger or your stupid adaptive UI. The screen is large enough to hold the current 3-4 toolbars. And I can't find a better UI. To go into a hamburger, to change the colour...

**Reply by LinuxGuy-cyber** [reply]
> I don't need a hamburger or your stupid adaptive UI. The screen is large enough to hold the current 3-4 toolbars. And I can't find a better UI. To go into a hamburger, to change the colour...

It feels good to know people who use logic still exist...

**Comment by JesusChapman** [comment]
Here you can see how the app that Libadwaita uses adapts to changes in its size and device .. and all with Libadwaita bookstores! Clean code, without complex processes. out-the-box:

https://github.com/user-attachments/assets/e60c3230-974c-4e90-a6e3-84c51406dc4d

**Reply by medmedin2014** [reply]
The real problem is how GNOME is maintaining those modified GTK widgets' library, their decisions are not at all inclusive for other Linux desktops, they do their stupid changes to fix and adjust things only for their GNOME desktop.

The most real example why LibAdwaita is failure for any complex application that will run on multiple Linux desktops and other platforms like Windows and Mac, is EasyEffects application when it became too much complex, and when the devs needed the real-time interaction and be real cross-platform they decided to port the whole application to Qt because GNOME devs are too much greedy and their ideas are too much narrow.

**Reply by Fighter19** [reply]
That's a weird thing to say, considering Qt froze widget development with version 5+
and decided to focus on QML. (At least so I heard from a Qt trainer like 8 years ago)
Let alone their, in my opinion, shady business model of letting you use their LGPL library, that they occasionally sprinkle with GPL modules, so you can't use it with 100% certainty in a commercial application, when it's not GPL.
To make you pay for their overpriced commercial license.

Like, obviously, when nothing is getting developed or changed, it's stable.
But at that point, you might as well use GTK3.

GNOME does interaction engineering, sometimes good sometimes bad, Qt is a framework.
That being said, I agree with GNOME not always making sense and making decisions I disagree with,
especially when it comes to touch screen usage, which isn't that unimportant for a software like Xournalpp.

* Removal of steppers (awful decision, accessibility and UX-wise)
* Scrollbar being tiny and requiring a long press to freely scroll (I tried telling that to customers, they almost killed me on the spot when they tested that!),
* Not being able to select multiple items on a multi selection list view, etc...
* Blurry font (according to clients, appears to be caused by FreeType, but is definitely not an issue in Windows with Qt, as that uses Microsoft's font rendering)
* New unasked-for GL renderer that simply broke rendering on older devices, for the sake of modernity.
(Instead of being like "just use Cairo" on machines that are actually strong enough, aka modern devices, they use that argument for weak legacy hardware).

In my opinion, you need to be able to use an UI completely drunk, and GTK does not fulfill that requirement with the newer versions.

**Comment by LinuxGuy-cyber** [comment]
Please don't use the libadwaita library.
It will limit Xournal++'s potential by forcing simplicity.

I sometimes fail to realise how GNOME users don't lose their sanity with the useless file manager (Nautilus). Once upon a time, it was a good file manager. They basically ruined it every way possible.

**Reply by carbon-starlight** [reply]
What features do you find lacking in Nautilus?

**Reply by bhennion** [reply]
This is clearly off-topic. This thread is already quite a mess. Please do not make it worse.
